### PR TITLE
RLink fixes

### PR DESCRIPTION
**Pull request by om26er**
This PR is supposed to fix the following
https://github.com/crossbario/crossbar/issues/1909
https://github.com/crossbario/crossbar/issues/1895
and partly https://github.com/crossbario/crossbar/issues/1894

### CODE REVIEW

**Comment by oberstet** [comment]
this is great! real progress rgd stabilizing rlinks, highly appreciated=)

**Comment by om26er** [comment]
This PR is now good to go.

File: crossbar/worker/rlink.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def on_subscription_delete(session_id, sub_id, details=None):
         self.log.debug("{me}: event forwarding setup done", me=self)
 
     @inlineCallbacks
-    def _setup_invocation_forwarding(self, other):
+    def _setup_invocation_forwarding(self, other: ApplicationSession):
```
Comment (oberstet) [inline crossbar/worker/rlink.py:210]: awesome! type hiiiints=)

File: crossbar/worker/rlink.py [hunk 2]
```diff
@@ -366,12 +373,41 @@ def on_registration_delete(session_id, reg_id, details=None):
...
+            yield register_current()
+        else:
+            # from the local leg, don't try to register procedures on the
+            # remote leg unless the remote session is established.
+            # This avoids issues where in-router components register procedures
+            # on startup and when the rlink is setup, the local leg tries to
+            # register procedures on the remote leg, even though the connection
+            # hasn't established.
+            # See: https://github.com/crossbario/crossbar/issues/1895
+            other.on('join', on_remote_join)
```
Comment (oberstet) [inline crossbar/worker/rlink.py:409]: yep, that's the bit that what missing! seems like you grok rlinks=)

**Pull request merged**